lib/zip/extra_field.rb

```diff
@@ -3,7 +3,7 @@ class ExtraField < Hash
     ID_MAP = {}
 
     def initialize(binstr = nil)
-      binstr and merge(binstr)
+      binstr && merge(binstr)
```

Comment:
Hi, I've been lurking here for a while, but thought I could answer this
question...

"&&" is usually used in its traditional sense, as a logical operator.
However, in other languages, it has some history of use as flow; being used
as a shorthand "if...then", or to replace the ternary operator. Ruby was
designed with the "and" keyword for those cases.

If this section was to be rewritten, a better way than using "&&" would be:

    merge(binstr) if binstr

The fact that it is more understandable rewritten as a conditional is the
biggest indication that the choice of "and" is better than "&&".

Hope this helps!

Mark
On Mar 23, 2015 8:56 AM, "Pavel Lobashov" <notifications@github.com> wrote:

> In lib/zip/extra_field.rb
> <https://github.com/rubyzip/rubyzip/pull/226#discussion_r26949762>:
>
> > @@ -3,7 +3,7 @@ class ExtraField < Hash
> >      ID_MAP = {}
> >
> >      def initialize(binstr = nil)
> > -      binstr and merge(binstr)
> > +      binstr && merge(binstr)
>
> @simonoff <https://github.com/simonoff> Can you help me to simplify this
> statement without usage of 'and', I read
> http://stackoverflow.com/questions/1426826/difference-between-and-and-in-ruby,
> but still not quite sure about it
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rubyzip/rubyzip/pull/226/files#r26949762>.
>